sources/packages/backend/apps/queue-consumers/src/services/student-assessment/student-assessment.service.ts

```diff
@@ -55,4 +60,42 @@ export class StudentAssessmentService {
       workflowName: data.workflowName,
     };
   }
+
+  /**
+   * Retrieve assessment cancellations to be retried up to a date.
+   * @param retryMaxDate max date to retrieve assessment cancellations to be retried.
+   * @returns student assessment cancellations to be retried.
+   */
+  async getAssessmentCancellationsRequestedToBeRetried(
+    retryMaxDate: Date,
+  ): Promise<StudentAssessment[]> {
+    return this.studentAssessmentRepo.find({
+      select: {
+        id: true,
+      },
+      where: {
+        studentAssessmentStatus: StudentAssessmentStatus.CancellationQueued,
+        updatedAt: LessThan(retryMaxDate),
+      },
+    });
+  }
+
+  /**
+   * Retrieve assessment start to be retried up to a date.
+   * @param retryMaxDate max date to retrieve assessment start to be retried.
+   * @returns student assessment start to be retried.
+   */
+  async getAssessmentQueuedToBeRetried(
```

Comment:
`getAssessmentCancellationsRequestedToBeRetried` and `getAssessmentQueuedToBeRetried` can potentially be converted to only on method accepting the studentAssessmentStatus as parameter defined as below.
```ts
studentAssessmentStatus: StudentAssessmentStatus.Queued | StudentAssessmentStatus.CancellationQueued
```